modules/ztp-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,220 @@
+
+//TODO Expand on what the seed image contains and explicitly say what it doesn't
+
+.Prerequisites
+
+* You deployed an {sno} with DU profile.
+* You installed the {lcao} on the seed cluster.
+* You installed the OADP Operator on the seed cluster.
+* You are logged in as a user with `cluster-admin` privileges.
+* The seed cluster has the same CPU topology as the target cluster.
+* The seed cluster and the target cluster have the same {rh-rhacm} or multicluster engine version.
+* The seed cluster is registered as a managed cluster.
+// Are there more prereqs?
+
+.Procedure
+
+. Detach the seed cluster.
+// TODO expand on this. Needs to be detached as LCA cleans up the ACM resources. If it remains detached to the hub,
+// the resources are recreated and will be included in the seed image.
+
+. Create the `Secret` CR in the `openshift-lifecycle-agent` namespace.
++
+--
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seedgen <1>
+  namespace: openshift-lifecycle-agent
+type: Opaque
+data:
+  seedAuth: <encoded_AUTHFILE> <2>
+  hubKubeconfig: <encoded_KUBECONFIG> <3>
+----
+<1> The `Secret` CR must have the `name: seedgen` and `namespace: openshift-lifecycle-agent` fields.
+<2> Specifies a base64-encoded authfile for write-access to the registry for pushing the generated seed images.
+<3> Optional. Specifies a base64-encoded `KUBECONFIG` file for administrator access to the hub cluster to unregister the seed cluster from {rh-rhacm}. If this is not present in the secret, the {rh-rhacm} cleanup is skipped and the seed cluster must be detached from the hub cluster manually.
+--
+
+.. Create the authentication and the `Kubeconfig` files by running the following commands:
++
+--
+.Authentication file
+[source,terminal]
+----
+$ MY_USER=myuserid
+$ AUTHFILE=/tmp/my-auth.json
+$ podman login --authfile ${AUTHFILE} -u ${MY_USER} quay.io/${MY_USER}
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${AUTHFILE} ; echo
+----
+
+.hubKubeconfig file
+[source,terminal]
+----
+$ HUBKUBECONFIGFILE=/home/<user_ID>/hubkubeconfig
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${HUBKUBECONFIGFILE} ; echo
+----
+--
+
+.. Copy the outputs into the respective `seedAuth` and `hubKubeconfig` fields in the `Secret` CR.
+
+. Apply the `Secret` CR.
++
+[source,terminal]
+----
+$ oc apply -f secretseedgenerator.yaml
+----
+
+. (Optional) If you did not populate the `hubKubeconfig` field in the `Secret` CR, detach the seed cluster from the hub cluster and remove all {rh-rhacm} and multicluster engine certificates that cannot be recertificated in the target cluster.
+
+.. Detach the seed cluster from the hub cluster:
++
+[source,terminal]
+----
+$ oc delete managedcluster sno-worker-example
+----
+
+.. Wait until the `ManagedCluster` CR is removed. Once the CR is removed, create the proper `SeedGenerator` CR. The {lcao} cleans up the {rh-rhacm} artifacts.
```

Comment:
somewhat outside of this update but if customer want to resize their cluster do we indicate when we recommend them to do it? before LAO starts, after update completed or somewhere here when managecluster is deleted?

Reply:
Can you elaborate on resize ?  This is a SNO.

Reply:
If it is just for SNO then I will drop my comment.